src/server/scripts/Northrend/IcecrownCitadel/instance_icecrown_citadel.cpp

```diff
@@ -1482,6 +1494,31 @@ class instance_icecrown_citadel : public InstanceMapScript
                 }
             }
 
+            void SetPositionTraps(GameObject* go)
+            {
+                std::vector<Position> trapPositions;
+
+                switch (go->GetEntry())
+                {
+                    case GO_SPIRIT_ALARM_1:
+                        trapPositions = GoSpiritAlarm_1;
+                        break;
+                    case GO_SPIRIT_ALARM_2:
+                        trapPositions= GoSpiritAlarm_2;
+                        break;
+                    case GO_SPIRIT_ALARM_3:
+                        trapPositions = GoSpiritAlarm_3;
+                        break;
+                    case GO_SPIRIT_ALARM_4:
+                        trapPositions = GoSpiritAlarm_4;
+                        break;
+                    default:
+                        break;
```

Comment:
maybe just ```return;``` in this case ? as the trapPositions will be an empty vector that go->Relocate() will try to use

Reply:
Done